qt-models/divetripmodel.cpp

```diff
@@ -376,6 +376,7 @@ void DiveTripModelBase::clear()
 {
 	beginResetModel();
 	clear_dive_file_data();
+	clearData();
```

Comment:
Ahhh - that was why things were crashing for me on mobile without disconnecting the model.
We were using the pointers in the items...

Reply:
Ultimately, there should be no reason to disconnect the model.

Reply:
So there were two distinct reasons:
(a) when applying the GPS fixes, the ListViews (both of them) didn't notice that there was a change in the dives and the GPS fixes wouldn't show up until after you restarted the app. I tried figuring out the data changed signals back then and couldn't. I think you have that addressed (at least I read that in one of the commits in one of your branches). I haven't tested that, yet.
(b) when changing dive files we were dereferencing bogus pointers that came from the items array. I think this should be fixed with this.